storage_service/locations/models/dspace_rest.py

```diff
@@ -0,0 +1,404 @@
+                raise ValueError(msg)
+
+            msg = 'ArchivesSpace Server error: {}'.format(e.response.text)
+            raise DSpaceRESTException(msg)
+
+    def _deposit_aip_to_dspace(self, source_path, headers, ds_item, ds_sessionid):
+        LOGGER.info('Package is AIP. Sending file.')
+
+        bitstream_url = self.ds_rest_url.scheme + '://' + self.ds_rest_url.netloc + self.ds_rest_url.path + '/items/'
+        bitstream_url += ds_item['uuid'] + '/bitstreams?name=' + urllib.quote(os.path.basename(source_path).encode('utf-8'))
+
+        try:
+            with open(source_path, 'rb') as content:
+                requests.post(bitstream_url,
+                              headers=headers,
+                              cookies={'JSESSIONID': ds_sessionid},
+                              data=content,
+                              verify=self.verify_ssl)
+            LOGGER.debug('%s successfully sent to %s', source_path, bitstream_url)
+        except Exception:
+            msg = 'Error sending {}, to {}'.format(source_path, bitstream_url)
+            LOGGER.error(msg)
+            raise DSpaceRESTException(msg)
+
+    def move_from_storage_service(self, source_path, destination_path, package=None):
+        LOGGER.info('source_path: %s, destination_path: %s, package: %s', source_path, destination_path, package)
+        LOGGER.info('Package UUID: %s', package.uuid)
+        LOGGER.info('Package type: %s', package.package_type)
+        LOGGER.info('Verify SSL: %s', self.verify_ssl)
+
+        if package is None:
+            LOGGER.error('DSpace requires package param')
+            return
+
+        # This only handles compressed AIPs
+        if package.package_type == 'AIP' and not os.path.isfile(source_path):
+            raise NotImplementedError(_('Storing in DSpace does not support uncompressed AIPs'))
+
+        self._parse_and_clean_urls()
+
+        # Item to be created in DSpace
+        metadata, destinations, package_title = self._get_metadata(source_path, package.uuid, package.package_type)
+
+        ds_collection, as_archival_obj = self._assign_destination(package.package_type, destinations)
+
+        ds_sessionid = self._login_to_dspace_rest()  # Logging in to REST api gives us a session id
+
+        # Headers for connecting to DSpace REST API
+        headers = {
+            'Accept': 'application/json',
+            "Content-Type": 'application/json',
+        }
+
+        try:
+            ds_item = self._create_dspace_record(headers, metadata, ds_sessionid, ds_collection)
+
+            LOGGER.info('DSpace item: %s', ds_item)
+
+            if package.package_type == 'DIP':
+                self._deposit_dip_to_dspace(source_path, headers, ds_item, ds_sessionid)
+
+                if self.as_url != '' and self.as_user != '' and self.as_password != '' and self.as_repository != '':
+                    self._link_dip_to_archivesspace(as_archival_obj, package.uuid, package_title, ds_item)
```

Comment:
It would be more idiomatic to express this condition as `if all([self.as_url, self.as_user, self.as_password, self.as_repository]):`, but more importantly I think it's possible for `as_archival_obj` to evaluate to `None` here, so I think it needs to be `if all([self.as_url, self.as_user, self.as_password, self.as_repository, as_archival_object]):`

Reply:
Spot on